exercises/danny_hindson/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,89 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+
+#include <algorithm>
+#include <cctype>
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <sstream>
+#include <string>
+#include <string.h>
+#include <vector>
+
+bool compare(const std::pair<std::string,int>& x, const std::pair<std::string,int>& y) {
```

Comment:
In the main code base we generally try to document functions/methods/classes using Doxygen as in an example [DataService](https://github.com/mantidproject/mantid/blob/master/Framework/Kernel/inc/MantidKernel/DataService.h#L420).

I would also consider defining an alias for `std::pair<std::string,int>` as it's used quite frequently over the code. In modern C++ this can be done as 

```
using WordCount = std::pair<std::string, int>;
```
This would go above the function definition. Giving it a name like this also serves to improve readability.

Reply:
Thanks. So are people just keying in comments of this general style by hand into the code units? Or if I install Doxygen do I get some kind of tool to support creating this kind of documentation?

Reply:
While doxygen itself doesn't provide a tool several IDEs will create a template documentation string using the parameter names in the function definitions. QtCreator can do this out of the box and Visual Studio can do it with a [plugin](https://marketplace.visualstudio.com/items?itemName=cschlosser.doxdocgen).